pkg/apiserver/handlers.go

```diff
@@ -197,7 +203,20 @@ func WithAuthorizationCheck(handler http.Handler, getAttribs RequestAttributeGet
 	})
 }
 
-// KindAndNamespace returns the kind, namespace, and path parts for the request relative to /{kind}/{name}
+// APIRequestInfo holds information parsed from the http.Request
+type APIRequestInfo struct {
+	// Verb is the kube verb associated with the request, not the http verb.  This includes things like list and watch.
+	Verb       string
+	APIVersion string
+	Namespace  string
+	Kind       string
```

Comment:
> This doesn't fix the multiple resource problem.

It's not intended to.  This is making the method more generally useful and usable without changing the semantics for existing callers.